File: check_layout.js
import puppeteer from 'puppeteer';

(async () => {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();
  await page.setViewport({ width: 1920, height: 1080 });
  await page.goto('http://localhost:3000/services', { waitUntil: 'networkidle0' });

  const info = await page.evaluate(() => {
    const main = document.querySelector('.quantum-page');
    const wrap = document.querySelector('.wrap');
    const body = document.body;
    
    return {
      body: {
        width: body.getBoundingClientRect().width,
        display: getComputedStyle(body).display
      },
      main: {
        width: main ? main.getBoundingClientRect().width : null,
        display: main ? getComputedStyle(main).display : null,
        marginLeft: main ? getComputedStyle(main).marginLeft : null
      },
      wrap: {
        width: wrap ? wrap.getBoundingClientRect().width : null,
        marginLeft: wrap ? getComputedStyle(wrap).marginLeft : null,
        display: wrap ? getComputedStyle(wrap).display : null
      }
    };
  });

  console.log(JSON.stringify(info, null, 2));
  await browser.close();
})();

File: check_layout.py
from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page(viewport={'width': 1920, 'height': 1080})
    page.goto('http://localhost:3000/services')
    
    html = page.evaluate('() => { const el = document.documentElement; const rect = el.getBoundingClientRect(); return { width: rect.width, display: getComputedStyle(el).display }; }')
    body = page.evaluate('() => { const el = document.body; const rect = el.getBoundingClientRect(); return { width: rect.width, display: getComputedStyle(el).display, flex_direction: getComputedStyle(el).flexDirection }; }')
    main = page.evaluate('() => { const el = document.querySelector("main"); const rect = el.getBoundingClientRect(); return { width: rect.width, display: getComputedStyle(el).display, margin: getComputedStyle(el).margin }; }')
    wrap = page.evaluate('() => { const el = document.querySelector(".wrap"); const rect = el.getBoundingClientRect(); return { width: rect.width, left: rect.left, margin: getComputedStyle(el).margin, max_width: getComputedStyle(el).maxWidth }; }')
    
    print(f'HTML: {html}')
    print(f'BODY: {body}')
    print(f'MAIN: {main}')
    print(f'WRAP: {wrap}')
    
    browser.close()
